tools/importers/gene_set_import.py

```diff
@@ -0,0 +1,174 @@
+    records = []
+
+    with open(filePath) as f:
+        # whether we're on the first line
+        thisLineColumns = not fields[0]["name"]
+
+        for line in f:
+            cells = line.split("\t")
+
+            # make sure it's the right size
+            assert len(cells) == len(fields)
+
+            # if it's a header line, parse the field names
+            if thisLineColumns:
+                thisLineColumns = False;
+
+                # TODO: parse the header row into columns
+
+                raise Exception("not implemented")
+                continue
+
+            # parse a data line
+            record = {
+                "associated_object": {
+                    "collection_name": "GeneSets",
+                    "mongo_id": geneSetId
+                }
+            }
+
+            # attach each of the column values to the
+            # record with the correct attributes
+            for i in range(len(fields)):
+                field = fields[i]
+                cell = cells[i]
+
+                # placeholder value to achieve scope
+                value = None
+
+                if field["value_type"] == "Number":
+                    if "." in cell:
+                        value = float(cell)
+                    else:
+                        value = int(cell)
+                elif field["value_type"] == "String":
+                    value = cell
+                else:
+                    print field
+                    raise Exception("invalid value type", field["value_type"])
+
+                record[field["name"]] = value
+
+            # add the record to be inserted later
+            records.append(record)
+
+    # insert the records for the gene set
+    db.records.insert_many(records)
+
+    # set up security for the new gene set
+    security = json.loads(securityString[1:-1])
+
+    securityAttribute = "associated_object"
+    if type(security) is list:
+        securityAttribute = "collaborations"
+
+    geneSet[securityAttribute] = security
+
+    # if metadata is specified, attach that too
+    if metadata:
+        geneSet["metadata"] = metadata
+
+    # add a couple more fields
+    geneSet["_id"] = geneSetId
+    geneSet["fields"] = fields
+
+    # create the gene set
+    db.gene_sets.insert(geneSet)
+
+def main():
+    argv = sys.argv
+
+    # need at least 8 arguments to be valid
+    if len(argv) > 8:
+        fields = []
+        metadata = {}
+
+        # parse the given information into fields
+        if "--fieldDefinitions" in argv:
+            startIndex = argv.index("--fieldDefinitions") + 1
+
+            fieldInfo = argv[startIndex:]
+            for i in range(len(fieldInfo) / 2):
+                print i
+                fields.append({
+                    "name": fieldInfo[i * 2],
+                    "value_type": fieldInfo[i * 2 + 1]
+                })
+                print fieldInfo[i * 2], fieldInfo[i * 2 + 1]
+        else:
+            # TODO
```

Comment:
- the fieldTypesOnly option appears to be NYI; however I don't see that it's actually ever used
- don't waste your time adding this now, but for future reference python has a csv library that can among other things create dicts using the first line as keys. see csv.dictReader